### DISCUSSION TITLE
Query join with supabase js SDK

### DISCUSSION THREAD

**Discussion by pr0t0n-cmd** [supabase/supabase#20674] [Questions]
I have a table "contracts", on this table i have a reference to **customer_id**.
I want to be able to select **first_name** and **last_name** from the "customers" table.
i have tried this, but it returns **null**.
const { data: contracts } = await supabase
		.from('contracts')
		.select(`
		id,
		customers ( first_name, last_name )`);

**Comment by GaryAustin1** [comment]
This almost always means you don't meet RLS on your customers table if you are not getting an error and that part of the response is null.  You have to meet select on all tables involved.
But you are also not checking error in the return and don't say what exactly is returning as null.

**Reply by pr0t0n-cmd** [reply]
if i select tables each on their own it wokrs fine with rls, maybe something happens if i join them? it just gives me null
i have now tried to create a function like this and call it from sdk const { data, error } = await supabase.rpc('join _customer_table')

select
  contracts.*,
  customers.first_name,
  customers.last_name
from
  contracts
inner join customers on
contracts.customer_id = customers.customer_id

It is working fine, i try to impersonate various users and itas working as expected.
When i call it from inside supabase dashboard it gives me all the records, its OK. When i try to call it from sdk it just return the first record

**Reply by GaryAustin1** [reply]
You should check the error coming back.
If you can access each table separately from the API then the join should work.
When you say it returns the first record do you mean it returns the first table part (id) for all the rows and not the customers parts?

Although I would expect nothing to return if you did not have it (and an error), do you have a foreign key reference on between the tables?  Just make sure that is what you meant by reference.

Are you use next.js for you client side?

**Reply by pr0t0n-cmd** [reply]
"Searched for a foreign key relationship between 'contracts' and 'customers' in the schema 'public', but no matches were found.

thats the error

**Reply by GaryAustin1** [reply]
Then you don't have a foreign key setup.  One of your tables needs to have a foreign key reference defined to the other.

**Reply by pr0t0n-cmd** [reply]
i have a relationship on contracts, there's a column customer_id that is linked to customers on the customer_id column

**Reply by GaryAustin1** [reply]
Can you show this?  The API does not think so.

**Reply by GaryAustin1** [reply]
You can also run this `NOTIFY pgrst, 'reload schema'` in the SQL editor if you changed that relationship or added it as sometimes the cache has to be reloaded if you make changes to foreign keys definitions.

**Reply by pr0t0n-cmd** [reply]
ok, it wont allow me to save it, it doesn't see the relationship. what could i check for?

**Reply by GaryAustin1** [reply]
You need to show your table structures.  I don't want to keep guessing if I can help it.  
So you did not have a foreign key relationship before?

**Reply by pr0t0n-cmd** [reply]
ok, i have checked the logs and the errors, probably it wasnt creating the relationship anymore because the column customer_id was not **primary** in the customers table, maybe i have inadvertently turned it off. thanks for the support, now it is working fine.

**Reply by GaryAustin1** [reply]
Well you don't show all the detail on your rpc but I imagine you set it to return A record.  You would need to return a SETOF RECORDS.  This can only be done with the SQL editor though.
